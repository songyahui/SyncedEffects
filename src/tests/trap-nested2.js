"use hopscript"

var hh = require("hiphop");

var prg = <hh.module A B C D>
  <hh.sequence>
    <hh.emit A/>
    <hh.trap U>
      <hh.sequence>
	<hh.trap T>
	  <hh.sequence>
	    <hh.exit U/>
	    <hh.emit B/>
	  </hh.sequence>
	</hh.trap>
	<hh.emit C/>
      </hh.sequence>
    </hh.trap>
    <hh.emit D/>
  </hh.sequence>
</hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "trapnested2");
