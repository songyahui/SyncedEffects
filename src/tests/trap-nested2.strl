module trapnested2:

output A, B, C, D;

emit A;
trap U in
    trap T in
        exit U;
        emit B
    end trap;
    emit C
end trap;
emit D

end module