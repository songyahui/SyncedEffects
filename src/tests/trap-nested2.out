trapnested2> ;
--- Output: A D
trapnested2> ;
--- Output:
